pyglmnet/pyglmnet.py

```diff
@@ -718,23 +718,36 @@ def fit(self, X, y):
         # Iterative updates
         for t in range(0, self.max_iter):
             if self.solver == 'batch-gradient':
+                if t > 1:
+                    gradnorm_prev = gradnorm
                 grad = _grad_L2loss(self.distr,
                                     alpha, self.Tau,
                                     reg_lambda, X, y, self.eta,
                                     beta)
                 # Converged if the norm(gradient) < tol
-                if (t > 1) and (np.linalg.norm(grad) < tol):
+                gradnorm = np.linalg.norm(grad)
+                logger.info('Grad norm: {}'.format(gradnorm))
+                if t > 1:
+                    if ((self.reg_lambda == 0 and gradnorm < tol) or
+                        (self.reg_lambda > 0 and
+                         np.abs(gradnorm - gradnorm_prev) < tol)):
```

Comment:
hmm ... not sure. Looks too hackish. I'll try to play around with this PR later. Don't merge yet !